woocommerce/payment-gateway/integrations/class-sv-wc-payment-gateway-integration-subscriptions.php

```diff
@@ -425,23 +429,25 @@ public function get_order( $order ) {
 
 
 	/**
-	 * Don't copy order-specific meta to renewal orders from the WC_Subscription
-	 * object. Generally the subscription object should not have any order-specific
-	 * meta (aside from `payment_token` and `customer_id`) as they are not
-	 * copied during the upgrade (see do_not_copy_order_meta_during_upgrade()), so
-	 * this method is more of a fallback in case meta accidentally is copied.
+	 * Don't copy order-specific meta to renewal orders from the {@see WC_Subscription} object.
+	 *
+	 * Generally the subscription object should not have any order-specific meta (aside from `payment_token` and `customer_id`)
+	 * as they are not copied during the upgrade (see do_not_copy_order_meta_during_upgrade()),
+	 * so this method is more of a fallback in case meta accidentally is copied.
 	 *
 	 * @since 4.1.0
-	 * @param array $order_meta order meta to copy
-	 * @return array
+	 *
+	 * @param array<mixed>|scalar $order_meta order meta to copy
+	 * @return array<mixed>|scalar
 	 */
 	public function do_not_copy_order_meta( $order_meta ) {
 
 		$meta_keys = $this->get_order_specific_meta_keys();
 
-		foreach ( $order_meta as $index => $meta ) {
+		foreach ( (array) $order_meta as $index => $meta ) {
 
-			if ( in_array( $meta['meta_key'], $meta_keys ) ) {
+			// this accounts for different versions of the Subscriptions filter running before and after WooCommerce Subscriptions 2.5
+			if ( in_array( $index, $meta_keys ) || ( isset( $meta['meta_key'] ) && in_array( $meta['meta_key'], $meta_keys ) ) || ( isset( $meta['_meta_key'] ) && in_array( $meta['_meta_key'], $meta_keys ) ) ) {
```

Comment:
This became quite a complicated (to read) conditional :thinking: 

* I wonder if for future generations, there might be safety in writing each piece on a separate line? Not sure if any linters would complain.

```suggestion
			if (
			        in_array( $index, $meta_keys )
			        || ( isset( $meta['meta_key'] ) && in_array( $meta['meta_key'], $meta_keys ) )
			        || ( isset( $meta['_meta_key'] ) && in_array( $meta['_meta_key'], $meta_keys ) )
			) {
```

* Having to check both `meta_key` and `_meta_key` looks quite unfortunate, are they really both in play here? I read through `WC_Subscriptions_Data_Copier::copy_data()` and `WC_Subscriptions_Data_Copier::apply_deprecated_filter()` - I could only see literal `meta_key` in code. Maybe underscore version in comments is just an example for an actual meta key, like `_paid_date` etc?

Reply:
https://github.com/godaddy-wordpress/wc-plugin-framework/pull/642/commits/b1418ce909946173f9596412186c68a50e2aed9d